components/slices/banner.vue

```diff
@@ -0,0 +1,231 @@
+<template>
+  <section :class="`${bannerWithBackgroundImageClass}`">
+    <div :class="['banner', { 'banner--with-image': hasImage }]">
+      <div class="banner__main">
+        <prismic-rich-text :field="title" class="banner__title" />
+        <prismic-rich-text :field="textContent" class="banner__content" />
+        <slot></slot>
+        <div class="banner__button-group">
+          <div v-for="(link, index) in links" :key="`item-${index}`">
+            <pix-link
+              v-if="!isVideo(link)"
+              :field="link.bannerlinkurl"
+              class="banner__button"
+              :class="videoClass(link)"
+            >
+              {{ link.bannerlinktext }}
+            </pix-link>
+            <template v-if="isVideo(link)">
+              <button
+                class="banner__button banner__button-video"
+                @click="openVideoModal()"
+              >
+                <fa :icon="fas.faPlayCircle" />
+                {{ link.bannerlinktext }}
+              </button>
+              <modal ref="modal" name="videoModal" height="auto">
+                <VideoSlice :video-url="link.bannerlinkurl" />
+              </modal>
+            </template>
+          </div>
+        </div>
+      </div>
+      <prismic-image v-if="hasImage" :field="imageUrl" />
+    </div>
+  </section>
+</template>
+
+<script>
+import { fas } from '@fortawesome/free-solid-svg-icons'
+import VideoSlice from './video'
+
+export default {
+  name: 'Banner',
+  components: { VideoSlice },
+  props: {
+    content: {
+      type: Object,
+      default: null,
+    },
+  },
+  computed: {
+    bannerWithBackgroundImageClass() {
+      const hasBackgroundImage =
+        this.content.primary.bannerbackground &&
+        this.content.primary.bannerbackground.url
+      return hasBackgroundImage ? 'banner-with-background' : ''
+    },
+    title() {
+      return this.content.primary.bannertitle
+    },
+    textContent() {
+      return this.content.primary.bannercontent
+    },
+    links() {
+      return this.content.items
+    },
+    hasImage() {
+      return (
+        this.content.primary.bannerimage && this.content.primary.bannerimage.url
+      )
+    },
+    imageUrl() {
+      return this.content.primary.bannerimage
+    },
+    fas() {
+      return fas
+    },
+  },
+  mounted() {
+    if (this.bannerWithBackgroundImageClass) {
```

Comment:
D'un point de vue compréhension, il serait mieux d'avoir un test genre :
```suggestion
    if (this.hasBackgorundImage) {
```